frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx

```diff
@@ -281,8 +284,94 @@ export function AnsibleAutomationsForm(props: {
     }
     return curator
   }
+
+  function setAnsibleSelections(value: any) {
+    const errors: any[] = []
+    let secret: SetStateAction<string> = ansibleSelection
+    if (Object.keys(value).length) {
+      supportedCurations.forEach((type) => {
+        const path = `ClusterCurator[0].spec.${type}.towerAuthSecret`
+        const testSecret = get(value, `${type}.towerAuthSecret`)
+        if (!!testSecret && testSecret !== ansibleSelection) {
+          if (ansibleCredentials.findIndex((cred) => get(cred, 'metadata.name') === testSecret) === -1) {
+            errors.push({ path, message: t('{{testSecret}} is not an existing ansible credential', { testSecret }) })
+          } else {
+            secret = testSecret
+          }
+        }
+      })
+      if (!errors.length) {
+        setAnsibleSelection(secret)
+      }
+    }
+    return errors
+  }
+
+  function setAnsibleInventorySelection(testInventory: any) {
+    const errors: any[] = []
+    let selectedInventory: SetStateAction<string> = ansibleInventory
+    if (testInventory !== ansibleInventory) {
+      if (!!testInventory && !ansibleTowerInventoryList.includes(testInventory)) {
+        errors.push({
+          path: `ClusterCurator[0].spec.inventory`,
+          message: t('{{testInventory}} is not an existing ansible inventory', { testInventory }),
+        })
+      } else {
+        selectedInventory = testInventory
+      }
+    }
+    if (!errors.length) {
+      setAnsibleInventory(selectedInventory)
+    }
+    return errors
+  }
+
+  function setCustomHook(type: string, preHook: boolean, value: any) {
+    const errors: any[] = []
+    if (Array.isArray(value)) {
+      value.forEach(({ type: jobType, name }, index) => {
+        const path = `ClusterCurator[0].spec.${type}.${preHook ? 'prehook' : 'posthook'}.${index}`
+        if (!['Job', 'Workflow'].includes(jobType)) {
+          errors.push({ path: `${path}.type`, message: t('Must be Job or Workflow') })
+        }
+        if (name && !AnsibleTowerJobTemplateList?.includes(name)) {
+          errors.push({ path: `${path}.name`, message: t('"{{name}}" is not an existing job', { name }) })
+        }
+      })
```

Comment:
We need to check the `AnsibleTowerWorkflowTemplateList` if the type is `Workflow`.